mycore-base/src/main/java/org/mycore/common/xsl/MCRParameterCollector.java

```diff
@@ -366,14 +360,88 @@ public static MCRParameterCollector getInstanceFromUserSession() {
         return job == null ? new MCRParameterCollector() : new MCRParameterCollector(job.getRequest());
     }
 
+    /**
+     * This method is used to clear the cache of the properties that are used in the XSL transformations.
+     * This should only be used in tests.
+     */
+    public static void clearCache(){
+        SavePropertiesCacheHolder.clear();
+    }
+
     public int hashCode() {
         if (modified) {
             int result = LOGGER.hashCode();
             //order of map should not harm result
+            result += SavePropertiesCacheHolder.getSafePropertiesCache().hashCode();
             result += parameters.entrySet().stream().mapToInt(Map.Entry::hashCode).sum();
             hashCode = result;
             modified = false;
         }
         return hashCode;
     }
+
+    private static final class SavePropertiesCacheHolder {
+        private final static AtomicInteger COMPUTED_HASH_CODE = new AtomicInteger(0);;
+        private static volatile Map<String, String> SAFE_PROPERTIES_CACHE;
+
+        static synchronized void clear() {
+            SAFE_PROPERTIES_CACHE = null;
+            COMPUTED_HASH_CODE.set(0);
+        }
+
+        static Map<String, String> getSafePropertiesCache() {
+            if (SAFE_PROPERTIES_CACHE == null) {
+                synchronized (SavePropertiesCacheHolder.class) {
+                    if (SAFE_PROPERTIES_CACHE == null) {
+                        SAFE_PROPERTIES_CACHE = initializeSafeProperties();
+                    }
+                }
+            }
+            return SAFE_PROPERTIES_CACHE;
+        }
+
+        static Map<String, String> initializeSafeProperties() {
+            Map<String, String> safeProperties = new ConcurrentHashMap<>();
+
+            MCRConfiguration2.addPropertyChangeEventLister((k) -> true, (k, old, _new) -> {
```

Comment:
if a change listener is added, it should also be removed on the clear() method. Save the returning UUID and call removePropertyChangeEventListener()